File: labs/poo/lab04/Java/src/Desenvolvedor.java
public class Desenvolvedor extends Funcionario {
    private String linguagem_principal;

    public Desenvolvedor(String linguagem_principal, String nome, int salarioB) {
        super(nome, salarioB);
        this.linguagem_principal = linguagem_principal;
    }

    public String projetos_realizados() {
        return "Projetos realizados na linguem: " + this.linguagem_principal;
    }

    public String getLinguagem_principal() {
        return linguagem_principal;
    }

    public void setLinguagem_principal(String linguagem_principal) {
        this.linguagem_principal = linguagem_principal;
    }

    @Override
    public String descricao() {
        return "Desenvolvedor: " + this.getNome() + "\n" +
                "Salario: " + this.getSalario_base() + "\n" +
                "Linguagem principal: " + this.linguagem_principal;
    }
}

File: labs/poo/lab04/cpp/Desenvolvedor.cpp
#include "Desenvolvedor.hpp"

Desenvolvedor::Desenvolvedor(string linguaguem_principal, string nome, int salario_base)
    : linguaguem_principal(linguaguem_principal), Funcionairo(nome, salario_base) {}

// Desenvolvedor::Desenvolvedor(string linguaguem_principal) : linguaguem_principal(linguaguem_principal) {}

string Desenvolvedor::projetos_realizados()
{
    return "Projetos entregues na linguagem: " + linguaguem_principal;
}

string Desenvolvedor::getLingaguemPrincipal() const
{
    return linguaguem_principal;
}

void Desenvolvedor::setLinguaguemPrincipal(const string &lingaguem_principal)
{
    this->linguaguem_principal = linguaguem_principal;
}

string Desenvolvedor::Descricao()
{
    return "Nome: " + getNome() + "\nSalario: " + to_string(getSalarioBase()) + "\nLinguagem principal: " + getLingaguemPrincipal();
}